app/controllers/uploads_controller.rb

```diff
@@ -12,22 +10,29 @@ def new
     @upload = new_upload(params[:csv_type])
     return if @upload.csv_type_check?
 
-    log_and_display_error(errors_for_alert([@upload]), 'Warning')
+    alert_and_log(@upload.errors.full_messages.join(', '))
     redirect_to dashboards_path
   end
 
   def create
     @upload = Upload.create(merged_params)
 
     begin
-      data = load_csv
-      display_failed_instances(data.failed_instances)
+      failed = load_csv.failed_instances
+
+      @upload.check_for_headers
+      headers = @upload.all_warnings
```

Comment:
Variable name "header_warnings" would be more descriptive for future maintainers.